test/e2e/azure_test.go

```diff
@@ -597,4 +597,55 @@ var _ = Describe("Workload cluster creation", func() {
 			})
 		})
 	})
+
+	Context("Creating clusters using clusterclass [REQUIRED]", func() {
+		It("with a single control plane node, one linux worker node, and one windows worker node", func() {
+			// use "cc" as spec name because natgw pip name exceeds limit.
```

Comment:
that's not great... what's the full name that's getting rejected? Should we file an issue to look into reducing the name length?

Reply:
Yeah this is a bit tricky since natgw name is generated based on the subnet name which itself is [generated](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/9c74c714c2a7837f545ef484c0a3460fa0aa3c65/azure/defaults.go#L139) based on the cluster name. `fmt.Sprintf("pip-%s-%s-natgw", clusterName, subnetName)`. Since cluster class adds 6 characters (`-`+`5 char hash`), it adds 12 extra characters. We could potentially remove cluster name from natgwip name, but not sure if we can guarantee uniqueness if the subnet name was not generated.

Reply:
we could remove cluster name and add a char hash to ensure uniqueness

Reply:
I think this needs more discussion, and could become hard to review if I include those changes here. I have created an [issue](https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/2330) to discuss this further. wdyt?